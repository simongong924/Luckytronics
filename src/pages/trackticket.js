import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class trackticket extends Component {
    constructor() {
        super();

        this.state = {
            trackingid: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e) {
        let target = e.target;
        let value = target.type === 'checkbox' ? target.checked : target.value;
        let name = target.name;

        this.setState({
          [name]: value
        });
    }
    handleClick() {
    this.setState(state => ({
      isToggleOn: !state.isToggleOn
    }));
  }

    handleSubmit(e) {
        e.preventDefault();
        //this.props.history.push('/ticktform');

        console.log('The form was submitted with the following data:');
        console.log(this.state);
    }

    render() {
        return (
        <div className="FormCenter">
            <form onSubmit={this.handleSubmit} className="FormFields">
            <div className="FormField">
                <label className="FormField__Label" htmlFor="trackingid">Tracking ID</label>
                <input type="trackingid" id="trackingid" className="FormField__Input" placeholder="Enter your tracking number" name="trackingid" value={this.state.trackingid} onChange={this.handleChange} />
              </div>

              <div className="FormField">
                  <button className="FormField__Button mr-20">Track</button> <Link to="/ticktform" className="FormField__Link">Create new ticket</Link></div>
                  
            </form>
          </div>
        );
    }
}

export default trackticket;