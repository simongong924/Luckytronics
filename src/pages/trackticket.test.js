import React from 'react';
import { shallow } from 'enzyme';
import trackticket from './trackticket';
import renderer from 'react-test-renderer';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<trackticket />',() => {
/*
it('has 2 inputs', () => {
const editor = shallow(<SignInForm />);
expect(editor.find('input').length).toEqual(2);
});
*/
it("snaps",() =>{
const component = renderer.create(
<trackticket />
);
let tree = component.toJSON();

expect(tree).toMatchSnapshot();
});


});
