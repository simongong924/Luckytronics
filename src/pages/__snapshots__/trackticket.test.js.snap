// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<trackticket /> snaps 1`] = `<trackticket />`;
